optuna/trial/_trial.py

```diff
@@ -589,6 +589,7 @@ def objective(trial):
             # Do nothing if already reported.
             # TODO(hvy): Consider raising a warning or an error.
             # See https://github.com/optuna/optuna/issues/852.
+            warnings.warn("this 'step' {} is already repoted.".format(step))
```

Comment:
```suggestion
            warnings.warn("this `step` {} is already repoted.".format(step))
```

Reply:
We can improve the warning message by clarifying what happens for users, like

```python
"The reported value is ignored because this `step` {} is already reported.".format(step)
```

Reply:
>  Could you add a test?

Of course.
 I would like to confirm that the test is checking of return value is None at the multiple calling?

Reply:
Sure! Probably no because the correct `report` also returns `None`. I think adding the following code after [`test_report`](https://github.com/optuna/optuna/blob/99484054fcec97eb530960a8612c3e6058ee5690/tests/trial_tests/test_trial.py#L573-L600) is enough.

```python
def test_report_warning() -> None:

    study = create_study()
    trial = study.ask()

    trial.report(1.23, 1)

    with pytest.warns(UserWarning):
        trial.report(1, 1)
```